packages/compiler/src/core/node-host.ts

```diff
@@ -41,3 +41,26 @@ export const NodeHost: CompilerHost = {
     return pathToFileURL(path).href;
   },
 };
+
+async function readUtf8File(path: string) {
+  const buffer = await readFile(path);
+  const len = buffer.length;
+  if (len >= 2 && buffer[0] === 0xfe && buffer[1] === 0xff) {
+    throw new InvalidEncodingError("UTF-16 BE");
+  }
+  if (len >= 2 && buffer[0] === 0xff && buffer[1] === 0xfe) {
+    throw new InvalidEncodingError("UTF-16 LE");
+  }
+  if (len >= 3 && buffer[0] === 0xef && buffer[1] === 0xbb && buffer[2] === 0xbf) {
+    // UTF-8 byte order mark detected
+    return buffer.toString("utf8", 3);
```

Comment:
I'm wondering if we should remove the skip bom logic in the scanner, though it doesn't do any harm to defend against it there too I guess. But it's mostly dead code now.

Reply:
yeah I was thinking about that. Feel like it doesn't hurt to have that and if someone wants to parse typespec but loading their own files, it will just work vs having to deal with it like we hav eto do with `JSON.parse`

Reply:
Yeah, good call. :)